File: auth-data-grpc-client/src/main/java/com/cartobucket/auth/data/services/grpc/mappers/ProfileMapper.java
package com.cartobucket.auth.data.services.grpc.mappers;

import com.cartobucket.auth.data.domain.Profile;
import com.cartobucket.auth.data.domain.ProfileType;
import com.google.protobuf.Struct;

import java.util.Collections;

public class ProfileMapper {
    public static Profile toProfile(Struct profile, ProfileType profileType) {
        var _profile = new Profile();
        _profile.setProfileType(profileType);
            _profile.setProfile(Profile.fromProtoMap(profile.getFieldsMap()));
        return _profile;
    }


}

File: auth-api/src/main/kotlin/com/cartobucket/auth/api/server/routes/mappers/ProfileMapper.kt
// (C)2024
package com.cartobucket.auth.api.server.routes.mappers

import com.cartobucket.auth.data.domain.Profile
import com.cartobucket.auth.data.domain.ProfileType
import com.cartobucket.auth.data.domain.User

object ProfileMapper {
    fun toProfile(
        user: User,
        profile: Map<String, Any?>,
    ): Profile =
        Profile(
            profileType = ProfileType.User,
            profile = profile as? Map<String, Any>,
            resource = user.id,
            authorizationServerId = user.authorizationServerId,
        )

    fun toProfile(profile: Map<String, Any?>): Profile =
        Profile(
            profileType = ProfileType.Application,
            profile = profile as? Map<String, Any>,
        )
}
